cpp/tests/experimental/spatial/pairwise_point_in_polygon_test.cu

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/iterator_factory.cuh>
+#include <cuspatial/experimental/pairwise_point_in_polygon.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/device_vector.hpp>
+
+#include <thrust/host_vector.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+#include <gtest/gtest.h>
+
+using namespace cuspatial;
+
+template <typename T>
+struct PairwisePointInPolygonTest : public ::testing::Test {
+ public:
+  rmm::device_vector<vec_2d<T>> make_device_points(std::initializer_list<vec_2d<T>> pts)
+  {
+    return rmm::device_vector<vec_2d<T>>(pts.begin(), pts.end());
+  }
+
+  rmm::device_vector<std::size_t> make_device_offsets(std::initializer_list<std::size_t> pts)
+  {
+    return rmm::device_vector<std::size_t>(pts.begin(), pts.end());
+  }
```

Comment:
Future TODOs: All these can be replaced with `make_device_vector` in `vector_factories.cuh`.